numba_cuda/numba/cuda/runtime/memsys.cu

```diff
@@ -0,0 +1,92 @@
+#include "memsys.cuh"
+
+namespace detail
+{
+  void __device__ check_memsys()
```

Comment:
I think this is a good idea - do we have things set up sufficiently correctly in numba-cuda such that this assertion can fire (e.g. if `debug=True` for a `@cuda.jit` dewcorator) when `TheMSys` is null?

Reply:
I don't think the tests currently cover a use case where the memsys is null. This check only fires when the NRT is enabled, but the memsys is not allocated. However currently we always promise memsys is allocated if NRT is enabled.